bin/factorial/index.js

```diff
@@ -0,0 +1,33 @@
+import { countFactorial } from "../../pages/functional/js/factorial.js";
+
+import { Factorial } from "../../pages/oop/js/Factorial.js";
+
+import { handleArgument } from "./handleArgument.js";
+
+const cliArguments = process.argv.slice(2);
+const { type, method, n } = handleArgument(cliArguments);
+let result = 0;
+
+if (!["oop", "functional"].includes(type)) {
+  throw new Error(`invalid type ${type}. must be "oop" or "functional"`);
+}
+
+if (!["loop", "recursive"].includes(method)) {
+  throw new Error(`invalid method ${method}. must be "loop" or "recursive"`);
+}
+
+if (typeof parseInt(n) != "number") {
+  throw new Error(`invalid number ${n}. n should be a number"`);
+}
```

Comment:
Put this validation code in the `handleArgument` file so this file will be focused on running the action/logic part.